src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php

```diff
@@ -757,60 +788,35 @@ public static function getGlobalType(string $var_id, int $codebase_analysis_php_
         }
 
         if ($var_id === '$_FILES') {
+            $str = Type::getString();
             $values = [
-                'name' => new Union([
-                    new TString(),
-                    new TNonEmptyList(Type::getString()),
-                ]),
-                'type' => new Union([
-                    new TString(),
-                    new TNonEmptyList(Type::getString()),
-                ]),
-                'size' => new Union([
-                    new TIntRange(0, null),
-                    new TNonEmptyList(Type::getInt()),
-                ]),
-                'tmp_name' => new Union([
-                    new TString(),
-                    new TNonEmptyList(Type::getString()),
-                ]),
-                'error' => new Union([
-                    new TIntRange(0, 8),
-                    new TNonEmptyList(Type::getInt()),
-                ]),
+                'name' => $str,
```

Comment:
should this change be included in this PR? Feels like it should be added for 4.x?

Reply:
I suppose this could be backported as well, just wanted to include it in this PR since I also introduced memoization for global types and wanted to avoid merge conflicts in this section of the VariableFetchAnalyzer.

Reply:
Though it would have to be changed a bit, here I re-use the string type for example, that can cause nasty issues in 4.x where unions aren't immutable